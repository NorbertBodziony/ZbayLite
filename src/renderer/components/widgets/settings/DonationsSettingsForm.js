import React, { useState } from 'react'
import PropTypes from 'prop-types'

import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import { FormControlLabel, Checkbox, TextField } from '@material-ui/core'

const styles = theme => ({
  container: {
    padding: theme.spacing(1),
    marginTop: theme.spacing(1)
  },
  addressField: {
    fontSize: 13,
    color: theme.palette.colors.black
  }
})

const checkAddress = (address, updateDonationAddress, setAddressStatus) => {
  const isValid = (/^t1[a-zA-Z0-9]{33}$|^ztestsapling1[a-z0-9]{75}$|^zs1[a-z0-9]{75}$/).test(address)
  if (isValid) {
    updateDonationAddress(address)
    setAddressStatus(true)
  } else {
    setAddressStatus(false)
  }
}

export const DonationsSettingsForm = ({
  classes,
  updateDonation,
  donationAllow,
  initialValues,
  updateDonationAddress
}) => {
  const [isAddressValid, setAddressStatus] = useState(!!initialValues.donationAddress)
  return (
    <Grid container direction={'row'} className={classes.container} spacing={2}>
      <Grid xs={12} item>
        <Grid item xs={12}>
          <Typography variant='body1'>Dontaion Recipient</Typography>
          <TextField
            name='donationAddress'
            fullWidth
            defaultValue={initialValues.donationAddress || ''}
            placeholder='Enter Zcash Address'
            helperText={isAddressValid || 'Please insert correct address'}
            error={!isAddressValid}
            onChange={(e) => checkAddress(e.target.value, updateDonationAddress, setAddressStatus)}
            InputProps={{ className: classes.field }}
          />
        </Grid>
      </Grid>
      <Grid xs={12} item>
        <FormControlLabel
          control={
            <Checkbox
              checked={donationAllow === 'true'}
              onChange={e => {
                updateDonation(e.target.checked)
              }}
              color='primary'
            />
          }
          label='Allow for sending a small donation to Zbay team'
        />
      </Grid>
    </Grid>
  )
}

DonationsSettingsForm.propTypes = {
  classes: PropTypes.object.isRequired,
  donationAllow: PropTypes.string,
  initialValues: PropTypes.object.isRequired,
  updateDonation: PropTypes.func.isRequired,
  updateDonationAddress: PropTypes.func.isRequired
}

export default withStyles(styles)(DonationsSettingsForm)
