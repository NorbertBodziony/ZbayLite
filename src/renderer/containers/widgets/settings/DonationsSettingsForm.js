import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import identityHandlers from '../../../store/handlers/identity'
import DonationsSettingsForm from '../../../components/widgets/settings/DonationsSettingsForm'
import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = state => {
  return {
    initialValues: {
      donationAddress: identitySelectors.donationAddress(state)
    },
    donationAllow: identitySelectors.donationAllow(state),
    shieldingTax: identitySelectors.shieldingTax(state)
  }
}

export const mapDispatchToProps = (dispatch, props) =>
  bindActionCreators(
    {
      updateDonation: identityHandlers.epics.updateDonation,
      setDonationAddress: identityHandlers.actions.setDonationAddress,
      setDonationAllow: identityHandlers.actions.setDonationAllow,
      setShieldingTax: identityHandlers.actions.setShieldingTax,
      updateShieldingTax: identityHandlers.epics.updateShieldingTax,
      updateDonationAddress: (address) => identityHandlers.epics.updateDonationAddress(address)
    },
    dispatch
  )

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DonationsSettingsForm)
