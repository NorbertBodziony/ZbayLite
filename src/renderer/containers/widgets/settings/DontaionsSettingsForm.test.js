/* eslint import/first: 0 */

import { mapDispatchToProps } from './DonationsSettingsForm'

describe('AccountSettingsForm', () => {
  it('will receive right actions', () => {
    const actions = mapDispatchToProps(x => x)
    expect(actions).toMatchSnapshot()
  })
})
