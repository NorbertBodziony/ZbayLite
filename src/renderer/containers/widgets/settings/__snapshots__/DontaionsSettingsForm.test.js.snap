// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSettingsForm will receive right actions 1`] = `
Object {
  "setDonationAddress": [Function],
  "setDonationAllow": [Function],
  "setShieldingTax": [Function],
  "updateDonation": [Function],
  "updateDonationAddress": [Function],
  "updateShieldingTax": [Function],
}
`;
